core/src/databases/table.rs

```diff
@@ -584,6 +585,19 @@ impl LocalTable {
 
         Ok(schema)
     }
+
+    pub async fn validate_csv_content(upsert_queue_bucket_path: &str) -> Result<TableSchema> {
+        let rows = Arc::new(
+            UpsertQueueCSVContent {
+                upsert_queue_bucket_path: upsert_queue_bucket_path.to_string(),
+            }
+            .parse()
+            .await?,
+        );
+
+        let schema = TableSchema::from_rows_async(rows).await?;
```

Comment:
I suppose this is where the spawn_blocking is ?

Reply:
I will make sure the buf reader is indeed buffered to avoid blocking the CPU while parsing

Reply:
Since I decided to download the file entirely in the end